### ISSUE TITLE
Panic when running in CircleCI: called `Result::unwrap()` on an `Err` value: ColorOutOfRange

### ISSUE THREAD

**Issue by oherrala** [RustSec/cargo-audit#5]
CircleCI might not have a colorful terminal. This is with CircleCI's ubuntu-14.04. Cargo audit works with same project ran from macOS.

```console
$ cargo audit
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ColorOutOfRange', /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libcore/result.rs:837
stack backtrace:
   1:     0x7f2c7f8fda3a - std::sys::imp::backtrace::tracing::imp::write::h3188f035833a2635
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2:     0x7f2c7f9002ff - std::panicking::default_hook::{{closure}}::h6385b6959a2dd25b
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:349
   3:     0x7f2c7f8ffefe - std::panicking::default_hook::he4f3b61755d7fa95
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:365
   4:     0x7f2c7f9007a7 - std::panicking::rust_panic_with_hook::hf00b8130f73095ec
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:553
   5:     0x7f2c7f9005e4 - std::panicking::begin_panic::h6227f62cb2cdaeb4
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:515
   6:     0x7f2c7f900559 - std::panicking::begin_panic_fmt::h173eadd80ae64bec
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:499
   7:     0x7f2c7f9004e7 - rust_begin_unwind
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:475
   8:     0x7f2c7f92c51d - core::panicking::panic_fmt::h3b2d1e30090844ff
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libcore/panicking.rs:69
   9:     0x7f2c7f823e37 - core::result::unwrap_failed::h356962afdab9fbfc
  10:     0x7f2c7f825ac0 - cargo_audit::notify::h7b4d7f4efc53788c
  11:     0x7f2c7f8267f6 - cargo_audit::main::hde81dc6245844138
  12:     0x7f2c7f9076ca - __rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libpanic_unwind/lib.rs:98
  13:     0x7f2c7f900ee6 - std::rt::lang_start::h65647f6e36cffdae
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:434
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panic.rs:351
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/rt.rs:57
  14:     0x7f2c7e384f44 - __libc_start_main
  15:     0x7f2c7f8235c8 - <unknown>
  16:                0x0 - <unknown>
$<2>
cargo audit returned exit code 101
```

**Comment by oherrala** [comment]
It's possible to reproduce this issue in macOS by setting `$TERM` environment variable to `vt100`:

```console
$ export TERM=vt100
$ cargo audit
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ColorOutOfRange', /Users/rustbuild/src/rust-buildbot/slave/stable-dist-rustc-mac/build/src/libcore/result.rs:837
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

I also noticed there's assumption that `$TERM` is present:

```console
$ unset TERM
$ cargo audit
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /Users/rustbuild/src/rust-buildbot/slave/stable-dist-rustc-mac/build/src/libcore/option.rs:323
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by tarcieri** [comment]
Yeah, lots of unwraps in the current code. I'm curious if I can reuse Cargo's `Shell` type to handle these interactions better:

https://github.com/rust-lang/cargo/blob/master/src/cargo/core/shell.rs

**Comment by tarcieri** [comment]
Fix released in cargo-audit 0.2.0.

It should now autodetect if colors are available and only use them if that's the case.

A `--color never` option has also been added to explicitly disable colors.

**Comment by oherrala** [comment]
👍 Good work. I tested 0.2.0 and it works.
